examples/file-reader/Main.hs

```diff
@@ -71,6 +76,22 @@ viewModel Model {..} = view
       , div_ [] [ text info ]
       ]
 
+#ifndef ghcjs_HOST_OS
```

Comment:
@dmjio 
This is defined both for ghcjs and ghc's js backend. 
You want something like this
https://github.com/ghcjs/jsaddle/blob/b72b4d01fdc186d2c7447beb7273ad13358ff593/jsaddle/src/Language/Javascript/JSaddle/Exception.hs#L21-L27

Reply:
https://github.com/ghc/ghc/blob/master/testsuite/tests/javascript/T23346.hs

Reply:
> @alexfmpe where do you see ghcjs_HOST_OS defined for the new JS backend?

Hmm? AFAICT cabal considers it `os(ghcjs)` for both old ghcjs and new js backend https://hackage.haskell.org/package/Cabal-syntax-3.14.1.0/docs/Distribution-System.html#v:Ghcjs 
unlike `impl(ghcjs)`
https://hackage.haskell.org/package/Cabal-syntax-3.14.1.0/docs/Distribution-Compiler.html#v:GHCJS

Reply:
that's.... less than ideal, but found a workaround https://github.com/dmjio/miso/issues/759